apptest/testcase.go

```diff
@@ -103,6 +106,124 @@ func (tc *TestCase) MustStartVminsert(instance string, flags []string) *Vminsert
 	return app
 }
 
+type vmcluster struct {
+	*Vminsert
+	*Vmselect
+	vmstorages []*Vmstorage
+}
+
+func (c *vmcluster) ForceFlush(t *testing.T) {
```

Comment:
I think it'd be better to expose Ingestion Method for cluster struct. 

 It allows to simplify data ingestion by eliminating need of `ForceFlush` method call. 

 In this case `PrometheusAPIV1ImportPrometheus` and any other ingestion method is blocking call. Which unblocks only when records we ingested.

 It also would be great to track number of ingested records. 

Because currently multiple calls for data ingestion doesn't respect previously ingested metrics and this check will be unaccurate :
```
app.waitUntilSent(t, wantRowsSentCount)
```

Reply:
I think this is a good addition, but this PR can be merged without it. We can make things simpler later.

Reply:
Per offline discussion, will be addressed in a separate PR.